lib/plugins/plugin/install.js

```diff
@@ -35,68 +37,61 @@ class PluginInstall {
-        }
-        return BbPromise.bind(this)
-          .then(this.pluginInstall)
-          .then(this.addPluginToServerlessFile)
-          .then(this.installPeerDependencies)
-          .then(() => {
-            const message = [
-              'Successfully installed',
-              ` "${this.options.pluginName}@${this.options.pluginVersion}"`,
-            ].join('');
-            this.serverless.cli.log(message);
-          });
-      });
+    await this.validate();
+    const plugins = await this.getPlugins();
+
+    const plugin = plugins.find((item) => item.name === this.options.pluginName);
+    if (!plugin) {
+      this.serverless.cli.log('Plugin not found in serverless registry, continuing to install');
+    }
+
+    await this.pluginInstall();
+    await this.addPluginToServerlessFile();
+    await this.installPeerDependencies();
+
+    const message = [
+      'Successfully installed',
+      ` "${this.options.pluginName}@${this.options.pluginVersion}"`,
+    ].join('');
+    this.serverless.cli.log(message);
   }
 
-  pluginInstall() {
+  async pluginInstall() {
     const servicePath = this.serverless.config.servicePath;
     const packageJsonFilePath = path.join(servicePath, 'package.json');
 
-    return fileExists(packageJsonFilePath)
-      .then((exists) => {
-        // check if package.json is already present. Otherwise create one
-        if (!exists) {
-          this.serverless.cli.log('Creating an empty package.json file in your service directory');
-
-          const packageJsonFileContent = {
-            name: this.serverless.service.service,
-            description: '',
-            version: '0.1.0',
-            dependencies: {},
-            devDependencies: {},
-          };
-          return fse.writeJson(packageJsonFilePath, packageJsonFileContent);
-        }
-        return BbPromise.resolve();
-      })
-      .then(() => {
-        // install the package through npm
-        const pluginFullName = `${this.options.pluginName}@${this.options.pluginVersion}`;
-        const message = [
-          `Installing plugin "${pluginFullName}"`,
-          ' (this might take a few seconds...)',
-        ].join('');
-        this.serverless.cli.log(message);
-        return this.npmInstall(pluginFullName);
-      });
+    const exists = await fileExists(packageJsonFilePath);
+    // check if package.json is already present. Otherwise create one
+    if (!exists) {
+      this.serverless.cli.log('Creating an empty package.json file in your service directory');
+
+      const packageJsonFileContent = {
+        name: this.serverless.service.service,
+        description: '',
+        version: '0.1.0',
+        dependencies: {},
+        devDependencies: {},
+      };
+      await fse.writeJson(packageJsonFilePath, packageJsonFileContent);
+    }
+
+    // install the package through npm
+    const pluginFullName = `${this.options.pluginName}@${this.options.pluginVersion}`;
+    const message = [
```

Comment:
I know it's not strictly part of `BbPromise` refactoring, but while we're at it - could you also simplify construction of that message? I don't think we need `join` here and putting it all in one line should be okay.